astropy/cosmology/parameter/_core.py

```diff
@@ -218,7 +218,7 @@ def __set__(self, cosmology, value):
             value.setflags(write=False)
 
         # Set the value on the cosmology
-        setattr(cosmology, self._attr_name, value)
+        object.__setattr__(cosmology, self._attr_name, value)
```

Comment:
I'd do this when you actually make `cosmology` a dataclass.

Reply:
Sounds good.

Reply:
Done.